analysis/subset_data/SubsetBam.wdl

```diff
@@ -0,0 +1,169 @@
+  call SplitBam {
+    input:
+    subset_bam = SubsetBamByTag.subset_bam,
+    barcode_tags_to_split_by = barcode_tags_to_split_by,
+    quality_tags_to_split_by = quality_tags_to_split_by,
+    separators_to_split_by = separators_to_split_by,
+    disk = (subset_bam_size * 2) + additional_disk # max split size would be 2 * subset size
+  }
+
+  # Revert bams into fastqs
+  scatter (index in range(length(SplitBam.split_bams))) {
+    String base = if (defined(final_fastq_basenames)) then select_first([final_fastq_basenames])[index] else basename(SplitBam.split_bams[index])
+    Int split_bam_size = ceil(size(SplitBam.split_bams[index], "GB"))
+
+    call SamToFastq {
+      input:
+        bam = SplitBam.split_bams[index],
+        output_basename = base,
+        disk = (split_bam_size * 2) + additional_disk  # fast q shouldn't be larger than the bam size
+    }
+  }
+}
+
+task SelectCells {
+  File aligned_bam
+  Int num_large_cells
+  Int num_small_cells
+  Int? min_reads_per_cell
+  String tag = "CB"
+
+  Int disk
+
+  command <<<
+    set -e
+
+    java -jar /root/picard.jar GroupReadsByTag \
+    I=${aligned_bam} \
+    O=histogram \
+    T=${tag}
+
+    # write largest num_large_cells into file
+    cat histogram | grep -v "#" | awk '{if($2>${default="100" min_reads_per_cell})printf("%s\t%s\n", $1, $2)}' | sort -k2 -n | tail -n+1 | tail -${num_large_cells} | cut -f1 >> tags_to_filter
+
+    # write smallest num_small_cells into file
+    cat histogram | grep -v "#" | awk '{if($2>${default="100" min_reads_per_cell})printf("%s\t%s\n", $1, $2)}' | sort -k2 -n | tail -n+1 | head -${num_small_cells} | cut -f1 >> tags_to_filter
+  >>>
+  runtime {
+    docker: "jsotobroad/gold_data:1.0"
+    disks: "local-disk " + disk + " HDD"
+  }
+  output {
+    Array[String] tag_values = read_lines("tags_to_filter")
+    File histogram = "histogram"
+  }
+}
+
+task SubsetBamByTag {
+  File aligned_bam
+  Array[String] tag_values
+  String tag = "CB"
+
+  Int disk
+
+  command {
+    java -jar /root/picard.jar FilterSamReads \
+    I=${aligned_bam} \
+    FILTER=includeTagValues \
+    O=selected_cells.bam \
+    T=${tag} \
+    TV=${sep=" TV=" tag_values}
+  }
+  runtime {
+    docker: "jsotobroad/gold_data:1.0"
+    disks: "local-disk " + disk + " HDD"
+  }
+  output {
+    File subset_bam = "selected_cells.bam"
+  }
+}
+
+
+task SplitBam {
+  File subset_bam
+  Array[String] barcode_tags_to_split_by
+  Array[String] quality_tags_to_split_by
+  Array[String]? separators_to_split_by
+  String separator = if defined(separators_to_split_by) then "S= " else ""
+
+  Int disk
+
+  command {
+    java -Xmx3g -jar /root/picard.jar SplitBamByTags \
+    T=${sep=" T=" barcode_tags_to_split_by} \
+    Q=${sep=" Q=" quality_tags_to_split_by} \
+    I=${subset_bam} \
+    O=split \
+    ${separator}${default="" sep=" S=" separators_to_split_by}
+  }
+  runtime {
+    docker: "jsotobroad/gold_data:1.0"
+    disks: "local-disk " + disk + " HDD"
+    memory: "7 GB"
```

Comment:
Apparently you get 7.5gb for the same cost (same VM type). 
You also get a spare thread in case picard's ability to use background threads to do decompression is general.